### PR TITLE
Slow Unittesting in Agent Asteroid

### PR DESCRIPTION
**Pull request by ohachimOs**
## Slow unittesting
One unittest was taking too much time because it was using an actual website to tests the exploits. This is not optimal because the responses might change over time, so they want always fit the scenario we are aiming to replicate (Too many redirects in this case). These requests also took a lot of time to be resolved.

## Fix
Mock both `requests.request` and `pysnmp.hlapi.getCmd`, while still simulating the `TooManyRedirects` behavior.
This was done by mocking requests to return `status_code=302`, and a `location: same_url` header.

## Where the test duration went up

Timeout was increased from 30 to 90
Which definitely contributed to the unittest duration increase

image

### CODE REVIEW

File: tests/asteroid_agent_test.py [hunk 1]
```diff
@@ -23,17 +25,37 @@ def testAsteroidAgent_whenExploitCheckDetectVulnz_EmitsVulnerabilityReport(
...
 def testAsteroidAgent_whenTooManyRedirects_doesNotCrash(
-    exploit_instance_with_report: Iterator[Type[definitions.Exploit]],
     asteroid_agent_instance: asteroid_agent.AsteroidAgent,
     agent_mock: list[m.Message],
+    mocker: plugin.MockerFixture,
+    requests_mock: requests_mock.Mocker,
 ) -> None:
     """Ensure that the agent does not crash when there are too many redirects."""
+
+    def response_callback(request: Any, context: Any) -> str:
```
Comment (amine3) [inline tests/asteroid_agent_test.py:35]: Where are the "Too many redirects" implemented in this test?

**Review by amine3 — DISMISSED** [review]

File: tests/asteroid_agent_test.py (hunk 1 above)
Comment (ohachimOs) [reply tests/asteroid_agent_test.py:35]: image
This is how requests procs redirects
```python
        context.headers = {"Location": request.url}
        context.status_code = 302
        return ""
```

Comment (ohachimOs) [reply tests/asteroid_agent_test.py:35]: The header location, is taking it to the same url over and over and over until whatever max redirect we have is exceeded

**Review by nmasdoufi-ol — COMMENTED** [review]
Github actions still take > 15min and it still didn't finish 50% of the unit tests.

**Comment by ohachimOs** [comment]
@nmasdoufi-ol yes, it's 5 min less than how it was, there is another unittest I should add mocking to

**Review by 3asm — APPROVED** [review]

**Pull request merged**